documentation/docs/scenarios/upload-to-transportrequest/Readme.md

```diff
@@ -0,0 +1,197 @@
+                // <JENKINS_URL>/job/<JOB_NAME>/<BUILD_NUMBER>/artifact/<DEPLOYABLE>. Nota bene: this format is not an Jenkins API.
+                // The build number can be retrieved during the build via ${currentBuild.number}
+                archiveArtifacts artifacts: <deployable>
+            }
+        }
+
+        // attaches the deployable to a transport request
+        stage('attach') {
+            steps {
+                transportRequestUploadFile script: this,
+                                           transportRequestId: '<TRANSPORT_REQUEST_ID>', // can be omitted if present inside a git commit message
+                                           applicationUrl: '<THE_URL_TO_THE_DEPLOYABLE_ACCORDING_TO_PUBLISH_STAGE>'
+            }
+        }
+    }
+}
+```
+
+#### Configuration (`.pipeline/config.yml`)
+
+This is a basic configuration example, which is also located in the sources of the project.
+
+```yaml
+general:
+  changeManagement:
+    type: 'RFC'
+    endpoint: 'the RFC endpoint' # e.g. example.com'
+    credentialsId: 'RFC' # The id under which the credentials are provided on Jenkins, defaults to 'CM'
+    rfc:
+      developmentInstance: '01' # needs to be adjusted
+      developmentClient: '001' # needs to be adjusted
+      docker:
+        image: '<imageId>' # the image needs to be built on user side. The corresponding id needs to be provided here.
+        options: []
+        envVars: {}
+        pullImage: true|false # true in case the image is provided by a company specific docker registry
+ 
+steps:
+    transportRequestUploadFile:
+        codePage: <the code page>, # e.g. 'Cp1252'
+        acceptUnixStyleLineEndings: true|false
+        applicationName: '/your/application/name'
+        applicationDescription: 'Application description'
+        abapPackage: '/abap/package'
+```
+
+### Upload via ODATA
+
+#### Jenkinsfile
+
+Following the convention for pipeline definitions, use a `Jenkinsfile` which resides in the root directory of your development sources.
+
+```groovy
+@Library('piper-lib-os') _
+
+
+pipeline {
+
+    agent any
+
+    stages {
+        stage("prepare") {
+            steps {
+                deleteDir()
+                checkout scm
+                setupCommonPipelineEnvironment script: this
+            }
+        }
+
+        stage('build') {
+            steps {
+                // what needs to be done here depends on the project, maybe zipping the sources is sufficient
+                mtaBuild script: this
+            }
+        }
+
+        // attaches the deployable to a transport request,
+        // in case we have a prior call to mtaBuild this step sets the deployable
+        stage('attach') {
+            steps {
+                transportRequestUploadFile script: this,
+                                           transportRequestId: '<TRANSPORT_REQUEST_ID>' // can be omitted if present inside a git commit message
```

Comment:
```suggestion
                                           transportRequestId: '<TRANSPORT_REQUEST_ID>' // This can be omitted if present inside a Git commit message
```